.travis.yml

```diff
@@ -1,21 +1,82 @@
 sudo: required
 dist: trusty
 
-# install the stable release of chef-dk.
 addons:
   apt:
     sources:
       - chef-stable-trusty
     packages:
       - chefdk
 
+# Don't `bundle install` which takes about 1.5 mins
+install: echo "skip bundle install"
+
+services: docker
+
+env:
+  matrix:
+  - INSTANCE=default-centos-6
+  - INSTANCE=default-centos-7
+  - INSTANCE=default-debian-7
+  - INSTANCE=default-debian-8
+  - INSTANCE=default-ubuntu-1204
+  - INSTANCE=default-ubuntu-1404
+  - INSTANCE=default-ubuntu-1604
+  - INSTANCE=native-centos-7
+  - INSTANCE=replication-centos-6
+  - INSTANCE=replication-centos-7
+  - INSTANCE=replication-debian-7
+  - INSTANCE=replication-debian-8
+  - INSTANCE=replication-ubuntu-1204
+  - INSTANCE=replication-ubuntu-1404
+  - INSTANCE=replication-ubuntu-1604
+  - INSTANCE=galera-centos-6
+  - INSTANCE=galera-centos-7
+  - INSTANCE=galera-debian-7
+  - INSTANCE=galera-debian-8
+  - INSTANCE=galera-ubuntu-1204
+  - INSTANCE=galera-ubuntu-1404
+  - INSTANCE=galera-ubuntu-1604
+  - INSTANCE=datadir-changed-centos-6
+  - INSTANCE=datadir-changed-centos-7
+  - INSTANCE=datadir-changed-debian-7
+  - INSTANCE=datadir-changed-debian-8
+  - INSTANCE=datadir-changed-ubuntu-1204
+  - INSTANCE=datadir-changed-ubuntu-1404
+  - INSTANCE=datadir-changed-ubuntu-1604
+  - INSTANCE=port-changed-centos-6
+  - INSTANCE=port-changed-centos-7
+  - INSTANCE=port-changed-debian-7
+  - INSTANCE=port-changed-debian-8
+  - INSTANCE=port-changed-ubuntu-1204
+  - INSTANCE=port-changed-ubuntu-1404
+  - INSTANCE=port-changed-ubuntu-1604
+  - INSTANCE=plugins-centos-6
+  - INSTANCE=plugins-centos-7
+  - INSTANCE=plugins-debian-7
+  - INSTANCE=plugins-debian-8
+  - INSTANCE=plugins-ubuntu-1204
+  - INSTANCE=plugins-ubuntu-1404
+  - INSTANCE=plugins-ubuntu-1604
+  - INSTANCE=no-binlog-centos-6
+  - INSTANCE=no-binlog-centos-7
+  - INSTANCE=no-binlog-debian-7
+  - INSTANCE=no-binlog-debian-8
+  - INSTANCE=no-binlog-ubuntu-1204
+  - INSTANCE=no-binlog-ubuntu-1404
+  - INSTANCE=no-binlog-ubuntu-1604
+
+before_script:
+  - sudo iptables -L DOCKER || ( echo "DOCKER iptables chain missing" ; sudo iptables -N DOCKER )
+  - eval "$(/opt/chefdk/bin/chef shell-init bash)" && chef gem install berkshelf
+  - /opt/chefdk/embedded/bin/chef --version
+  - /opt/chefdk/embedded/bin/cookstyle --version
+  - /opt/chefdk/embedded/bin/foodcritic --version
+
+script: KITCHEN_LOCAL_YAML=.kitchen.dokken.yml /opt/chefdk/embedded/bin/kitchen verify ${INSTANCE}
```

Comment:
Fairly common to see this out there, without it your matrix would end up running both scripts for every env - by adding an additional env and overriding the script you're able to create a single job for style/spec.  HTH.